packages/webapp/pages/posts/[id].tsx

```diff
@@ -47,42 +47,49 @@ export const getSeoDescription = (post: Post): string => {
 };
 export interface Props {
   id: string;
-  postData?: PostData;
+  initialData?: PostData;
 }
 
 interface PostParams extends ParsedUrlQuery {
   id: string;
 }
 
-const PostPage = ({ id, postData }: Props): ReactElement => {
+const PostPage = ({ id, initialData }: Props): ReactElement => {
   const [position, setPosition] =
     useState<CSSProperties['position']>('relative');
   const { tokenRefreshed } = useContext(AuthContext);
   const router = useRouter();
   const { isFallback } = router;
 
-  if (!isFallback && !id) {
-    return <Custom404 />;
-  }
-
   const {
-    data: postById,
+    data: fetchedPost,
     isLoading,
     isFetched,
   } = useQuery<PostData>(
     ['post', id],
     () => request(`${apiUrl}/graphql`, POST_BY_ID_QUERY, { id }),
-    { initialData: postData, enabled: !!id && tokenRefreshed },
+    { initialData, enabled: !!id && tokenRefreshed, retry: false },
   );
 
+  // Need this as sometimes client side might still be loading, but we already have initial data.
+  const post = fetchedPost ?? initialData;
+
+  if ((!isFallback && !id) || (isFetched && !post?.post.title)) {
```

Comment:
Why do we check the title?

Reply:
Because we always have the ID from the param, which could technically be a false one.